### PR TITLE
Correct addmetadata item PUT rest endpoint for header 'content-type: …

### PR DESCRIPTION
**Pull request by mgrauer**
…application/json'

Cleans up the addmetadata endpoint and test to require (and work when) the client passes the metadata list as the request body and specifies a request header of "content-type: application/json".

### CODE REVIEW

File: modules/api/controllers/components/ApiComponent.php [hunk 1]
```diff
@@ -776,6 +776,8 @@ public function itemSetmultiplemetadata($args)
      *
      * @param token Authentication token
      * @param itemid The id of the item
+     * @param metadata The metadata list to add or update, should be passed in the request body
```
Comment (jamiesnape) [inline modules/api/controllers/components/ApiComponent.php]: `must be passed`?

File: core/controllers/components/ApiitemComponent.php [hunk 2]
```diff
@@ -179,6 +179,8 @@ public function itemSetmultiplemetadata($args)
      * @path /item/addmetadata/{id}
      * @http PUT
      * @param id The id of the item
+     * @param metadata The metadata list to add or update, should be passed in the request body
```
Comment (jamiesnape) [inline core/controllers/components/ApiitemComponent.php]: `must be passed`?

**Comment by mgrauer** [comment]
fixed, thanks for the QA.

**Comment by jamiesnape** [comment]
LGTM.

**Pull request merged**